### ISSUE TITLE
What's the difference with CrossEntropy and LogSigmoid in reward modeling?

### ISSUE THREAD

**Issue by Reason-Wang** [huggingface/trl#1167]
The loss in trl reward trainer is log sigmoid loss: https://github.com/huggingface/trl/blob/6614b8aa6bd752bf19963a5b308f44e079e8e9fe/trl/trainer/reward_trainer.py#L229-L232. However, in another project a cross entropy or mse loss is used and only one sample is used for one forward pass: https://github.com/lucidrains/PaLM-rlhf-pytorch/blob/e624afd8860d6df8f7a429dbe518a2fd40b4cc34/palm_rlhf_pytorch/reward.py#L124-L127. What is the difference and what loss should I use? Is there any advantage of log sigmoid loss?

**Comment by lvwerra** [comment]
See this issue here for an explanation (or refer to the Llama2 paper): https://github.com/huggingface/trl/issues/718
